packages/frontend/src/app/app-component/config/config.component.spec.ts

```diff
@@ -66,4 +70,24 @@ describe('ConfigComponent', () => {
     expectedTokenDisplayOption = !expectedTokenDisplayOption;
     expect(component.showToken).toEqual(expectedTokenDisplayOption);
   });
+
+  //test the export csv file button
+  it('should call the appropriate download link', async () => {
+    component.showAdminView = false;
+    component.orgName = 'testOrg';
+    component.repoName = 'testRepo';
+    component.envUrl = environment.baseUrl;
+    await fixture.detectChanges();
+
+    //click the export csv button
+    const downloadButton = fixture.debugElement.nativeElement.querySelector(
+      '.download-button'
+    );
+    downloadButton.click();
+    fixture.whenStable().then(() => {
+      expect(location.path()).toEqual(
```

Comment:
I'm sorry that test is incorrect. I meant to convert to a draft, I am just trying to get this test to work then should be good

Reply:
😅 According to a certain human, Magic is just unknown science. I don't think the harness is that powerful though.
To test the external redirection i think it can help to just extract the `window.location` somewhere like [here](https://github.com/GoogleCloudPlatform/flaky-service/pull/70/files#diff-43ee95303c542368da43e43342356dc4R39).